File: 0438 - Find All Anagrams in a String/solution.cpp
class Solution {
public:
    vector<int> findAnagrams(string s, string p) {
        int n = p.size(), val = 0;
        vector<int> res, t(26);
        for(auto i : p) if(t[i-'a']++ == 0) val++;
        for(int i = 0; i < s.size(); i++){
            if(i >= n) if(t[s[i-n]-'a']++ == 0) val++;
            if(--t[s[i]-'a'] == 0) val--;
            if(val == 0) res.push_back(i-n+1);
        }
        return res;
    }
};

File: 0438 - Find All Anagrams in a String/solution.py
class Solution:
    def findAnagrams(self, s: str, p: str) -> List[int]:
        res, hs, hp, n = [], [0]*26, [0]*26, len(p)
        for i in range(n):
            hp[ord(p[i])-ord('a')] += 1
        for i in range(len(s)):
            if i>=n: hs[ord(s[i-n])-ord('a')] -= 1
            hs[ord(s[i])-ord('a')] += 1
            if hs==hp: res.append(i-n+1)
        return res